index.js

```diff
@@ -30,7 +31,12 @@ Gulp.prototype.watch = function(glob, opt, task) {
     fn = this.parallel(task);
   }
 
-  return vfs.watch(glob, opt, fn);
+  var watcher = chokidar.watch(glob, opt);
+  if (fn) {
+    watcher.on('all', fn);
```

Comment:
This will include `addDir` and `unlinkDir` events. Listening to the 3 events explicitly is a better match with what was there before.